iceoryx_utils/test/moduletests/test_cxx_method_callback.cpp

```diff
@@ -0,0 +1,222 @@
+{
+    ConstMethodCallback<void> sut(&m_testClass, &TestClass::constVoidVoidMethod);
+    ConstMethodCallback<void> sut2(&m_testClass, &TestClass::constVoidVoidMethod);
+
+    EXPECT_TRUE(sut == sut2);
+    EXPECT_FALSE(sut != sut2);
+}
+
+TEST_F(MethodCallback_test, TwoCallbacksWithSameClassAndMethodAreEqual)
+{
+    MethodCallback<void> sut(&m_testClass, &TestClass::voidVoidMethod);
+    MethodCallback<void> sut2(&m_testClass, &TestClass::voidVoidMethod);
+
+    EXPECT_TRUE(sut == sut2);
+    EXPECT_FALSE(sut != sut2);
+}
+
+TEST_F(MethodCallback_test, TwoConstCallbacksWithWithDifferentClassPtrAreNotEqual)
+{
+    ConstMethodCallback<void> sut(&m_testClass, &TestClass::constVoidVoidMethod);
+    ConstMethodCallback<void> sut2(&m_testClass2, &TestClass::constVoidVoidMethod);
+
+    EXPECT_FALSE(sut == sut2);
+    EXPECT_TRUE(sut != sut2);
+}
+
+TEST_F(MethodCallback_test, TwoCallbacksWithWithDifferentClassPtrAreNotEqual)
+{
+    MethodCallback<void> sut(&m_testClass, &TestClass::voidVoidMethod);
+    MethodCallback<void> sut2(&m_testClass2, &TestClass::voidVoidMethod);
+
+    EXPECT_FALSE(sut == sut2);
+    EXPECT_TRUE(sut != sut2);
+}
+
+TEST_F(MethodCallback_test, TwoConstCallbacksWithWithDifferentMethodPtrAreNotEqual)
+{
+    ConstMethodCallback<void> sut(&m_testClass, &TestClass::constVoidVoidMethod);
+    ConstMethodCallback<void> sut2(&m_testClass, &TestClass::constVoidVoidMethod2);
+
+    EXPECT_FALSE(sut == sut2);
+    EXPECT_TRUE(sut != sut2);
+}
+
+TEST_F(MethodCallback_test, TwoCallbacksWithWithDifferentMethodPtrAreNotEqual)
+{
+    MethodCallback<void> sut(&m_testClass, &TestClass::voidVoidMethod);
+    MethodCallback<void> sut2(&m_testClass2, &TestClass::voidVoidMethod2);
+
+    EXPECT_FALSE(sut == sut2);
+    EXPECT_TRUE(sut != sut2);
+}
+
+TEST_F(MethodCallback_test, InvalidConstCallbackCalledWillReturnError)
+{
+    ConstMethodCallback<void> sut(static_cast<TestClass*>(nullptr), &TestClass::constVoidVoidMethod);
+
+    auto result = sut();
+    ASSERT_TRUE(result.has_error());
+    EXPECT_EQ(result.get_error(), MethodCallbackError::UNABLE_TO_CALL_METHOD_ON_NULLPTR_CLASS_PTR);
+}
+
+TEST_F(MethodCallback_test, InvalidCallbackCalledWillReturnError)
+{
+    MethodCallback<void> sut(static_cast<TestClass*>(nullptr), &TestClass::voidVoidMethod);
+
+    auto result = sut();
+    ASSERT_TRUE(result.has_error());
+    EXPECT_EQ(result.get_error(), MethodCallbackError::UNABLE_TO_CALL_METHOD_ON_NULLPTR_CLASS_PTR);
+}
+
+TEST_F(MethodCallback_test, ValidConstCallbackReturnsValue)
+{
+    ConstMethodCallback<int, int, int> sut(&m_testClass, &TestClass::myConstMethod);
+
+    auto result = sut(4, 5);
+    ASSERT_FALSE(result.has_error());
+    EXPECT_EQ(*result, 4 * 5);
+}
+
+TEST_F(MethodCallback_test, ValidCallbackReturnsValue)
+{
+    MethodCallback<int, int, int> sut(&m_testClass, &TestClass::myMethod);
+
+    auto result = sut(6, 7);
+    ASSERT_FALSE(result.has_error());
+    EXPECT_EQ(*result, 6 + 7);
+}
+
```

Comment:
tests for move c'tor and assignment operator are missing

Reply:
added tests